doc/design/fork.md

```diff
@@ -0,0 +1,30 @@
+
+ The shared address space also causes problems for the copying of the parent stack into the child. The new child stack has all the frame pointers updated such that a stack walk can be achieved in the current stack, any stack variables that may point to other stack variables on the same stack will be incorrect and will point to the parents stack. One  way to achieve this is to search the space on the stack where function parameters and stack variables reside and fix them up, but this is potentially error prone and may cause fixup of other random values on the stack. Therefore these stack pointers are not currently being fixed up.
+
+ Another problem with shared address space is related to the return of the fork() command. The parent may allocate some memory that will be used in teh child forked process, but when fork() returns to continue execution it may delete this memory, causing the child forked process to use memory after it is freed. For this reason the safest model may be to only support a strict fork/exec model where the parent waits for the child to execute a new process through one of the exec*() APIs. This mode of operation is enabled through the fork configuration.
+
+Because the fork syscall is not a typical implementation the application needs to opt-in to use it. This is done with a new entry in the application config.json that is used for signing and packaging, or via the command line for non-signed testing.
+
+| Name | Value |
+| -- | -- |
+| ForkMode | Enable different modes of operation for the fork() syscall. <br> **pseudo_wait_for_children**: (not yet implemented) Creates fork sharing parent address space and the parent process will wait for all forked children to shutdown before shutting down itself. <br> **pseudo_kill_children** - similar to the previous except when the parent is shutting down it will send a SIGKILL to all child fork processes and wait for all of them to exit. <br> **pseudo_wait_for_exec** - (not yet implemented) the parent process will wait for the child forked process to call exec*() or exit() before continuing. <br> **none** - fork is disabled and returns a ENOTSUPP error if called.|
```

Comment:
Is there a corresponding ``--fork-mode`` option for ``myst exec``?

Reply:
no because we can always use `--app-config-path config.json`

Reply:
Hmm? So far we have a complete set of options to avoid use of config.json.

Reply:
ok. adding